posthog/queries/person_query.py

```diff
@@ -83,73 +83,72 @@ def get_query(
             f", argMax({column_name}, version) as {alias}" for column_name, alias in self._get_fields()
         )
 
-        grouped_person_filters, grouped_person_params = self._get_grouped_person_filters(
-            prepend=f"grouped_filters_{prepend}"
-        )
-        person_filters, person_params = self._get_person_filters(prepend=prepend)
-        cohort_filters, cohort_filter_params = self._get_cohort_filters(prepend=prepend)
-
-        cohort_query, cohort_params = self._get_cohort_query()
+        (
+            person_filters_prefiltering_condition,
+            person_filters_finalization_condition,
+            person_filters_params,
+        ) = self._get_person_filter_clauses(prepend=prepend)
+        multiple_cohorts_condition, multiple_cohorts_params = self._get_multiple_cohorts_clause(prepend=prepend)
+        single_cohort_join, single_cohort_params = self._get_fast_single_cohort_clause()
         if paginate:
-            limit_offset, limit_params = self._get_limit_offset()
+            order = "ORDER BY max(created_at) DESC, id" if paginate else ""
+            limit_offset, limit_params = self._get_limit_offset_clause()
         else:
-            limit_offset = ""
-            limit_params = {}
-        search_clause, search_params = self._get_search_clause(prepend=prepend)
-        distinct_id_clause, distinct_id_params = self._get_distinct_id_clause()
-        email_clause, email_params = self._get_email_clause()
-        filter_future_persons_query = (
-            "and argMax(created_at, version) < now() + interval '1 day'" if filter_future_persons else ""
+            order = ""
+            limit_offset, limit_params = "", {}
+        search_prefiltering_condition, search_finalization_condition, search_params = self._get_search_clauses(
+            prepend=prepend
         )
-        updated_after_clause, updated_after_params = self._get_updated_after_clause()
+        distinct_id_condition, distinct_id_params = self._get_distinct_id_clause()
+        email_condition, email_params = self._get_email_clause()
```

Comment:
For future consideration: You might want to do the same with the email filter, as I suspect this will timeout as well otherwise.

Reply:
The email filter is super legacy AFAIK, I don't think it's used anywhere (though [I've not yet been able to verify what's the level of API use](https://posthog.slack.com/archives/C02E3BKC78F/p1683818907581189))